storage/src/linear/filebacked.rs

```diff
@@ -218,7 +218,7 @@ impl WritableStorage for FileBacked {
     ) -> Result<(), FileIoError> {
         let mut guard = self.cache.lock().expect("poisoned lock");
         for (addr, node) in nodes {
-            guard.put(*addr, node.clone());
+            guard.put(LinearAddress::new((*addr).get()).unwrap(), node.clone());
```

Comment:
Not sure why this got complicated, we're just moving the addr, and it implements Copy.

Reply:
Would like this to work
```suggestion
            guard.put(guard.put(addr.into(), node.clone()););
```

To get this to work, you'd need to `impl From<NonZeroU64> for LinearAddress` and change the loop to `for (&addr, node)`.